src/reducers/Material.js

```diff
@@ -104,6 +106,12 @@ function materialsSetBoundaryConditionsForOne(state, action) {
     return materialsUpdateOne(state, Object.assign(action, {material: newMaterial}));
 }
 
+export function materialsSetIsNonPeriodicForOne(state) {
+    const newMaterial = state.materials[state.index].clone();
+    state.materials[state.index].isNonPeriodic = !newMaterial.isNonPeriodic;
+    return materialsUpdateOne(state, Object.assign(newMaterial, {material: state.materials[state.index]}));
```

Comment:
Should be just like the `materialsSetBoundaryConditionsForOne` and others above:

```js
const newMaterial = state.materials[state.index].clone();
newMaterial.isNonPeriodic = !newMaterial.isNonPeriodic;
return materialsUpdateOne(state, Object.assign(action, {material: newMaterial}));
```